docker/docker.go

```diff
@@ -0,0 +1,178 @@
+}
+
+// Destroy stops and removes the docker container with the given id.
+func Destroy(id string) error {
+	d := deputy.Deputy{
+		Errors: deputy.FromStderr,
+	}
+	cmd := execCommand("docker", "stop", id)
+	if err := d.Run(cmd); err != nil {
+		return fmt.Errorf("error while stopping container %q: %s", err)
+	}
+
+	cmd = execCommand("docker", "rm", id)
+	if err := d.Run(cmd); err != nil {
+		return fmt.Errorf("error while removing container %q: %s", err)
+	}
+	return nil
+}
+
+// launchArgs converts the Process struct into arguments for the docker run
+// command.
+func launchArgs(p charm.Process) ([]string, error) {
+	if err := p.Validate(); err != nil {
+		return nil, fmt.Errorf("invalid proc-info: %s", err)
+	}
+
+	args := []string{"run", "--detach", "--name", p.Name}
+	for k, v := range p.EnvVars {
+		args = append(args, "-e", k+"="+v)
+	}
+
+	for _, p := range p.Ports {
+		// TODO(natefinch): update this when we use portranges
+		args = append(args, "-p", fmt.Sprintf("%d:%d/%s", p.External, p.Internal, "tcp"))
+	}
+
+	for _, v := range p.Volumes {
+		args = append(args, "-v", fmt.Sprintf("%s:%s:%s", v.ExternalMount, v.InternalMount, v.Mode))
+	}
+
+	// Image and Command must come after all options.
+	args = append(args, p.Image)
+	if p.Command != "" {
+		// TODO(natefinch): update this when we make command a list of strings
+		args = append(args, strings.Fields(p.Command)...)
+	}
+	return args, nil
+}
+
+// status is the struct that contains the schema returned by docker's inspect command
+type status struct {
+	State state
+	Name  string
+}
+
+type state struct {
+	Running    bool
+	Paused     bool
+	Restarting bool
+	OOMKilled  bool
+	Dead       bool
+	Pid        int
+	ExitCode   int
+	Error      string
+}
+
+// brief returns a short summary for the status.
+func (s *status) brief() string {
+	switch {
+	case s.State.Running:
+		return "Running"
+	case s.State.OOMKilled:
+		return "OOMKilled"
+	case s.State.Dead:
+		return "Dead"
+	case s.State.Restarting:
+		return "Restarting"
+	case s.State.Paused:
+		return "Paused"
+	}
+	return "Unknown"
+}
+
+// inspect calls docker inspect and returns the unmarshaled json response.
+func inspect(id string) (status, error) {
+	cmd := execCommand("docker", "inspect", id)
+	out := &bytes.Buffer{}
+	cmd.Stdout = out
+	d := deputy.Deputy{
+		Errors: deputy.FromStderr,
+	}
+	if err := d.Run(cmd); err != nil {
+		return status{}, err
+	}
+	return statusFromInspect(id, out.Bytes())
+}
+
+func statusFromInspect(id string, b []byte) (status, error) {
+	var st []status
+	if err := json.Unmarshal(b, &st); err != nil {
+		return status{}, fmt.Errorf("can't decode response from docker inspect %s: %s", id, err)
+	}
+	if len(st) == 0 {
+		return status{}, errors.New("no status returned from docker inspect " + id)
+	}
+	if len(st) > 1 {
+		return status{}, errors.New("multiple status values returned from docker inspect " + id)
+	}
+	return st[0], nil
+
+}
+
+// These two structs are copied from juju/process/plugin
```

Comment:
As we talked about, the issue of versioning needs more discussion.  Would you mind putting a TODO comment here about finding a better way than copying the structs?